[CI] ESVectorUtilTests testSoarDistance failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9520 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ndaocguq5fcqo)
- [elasticsearch-periodic-platform-support #9492 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/6u2ph6jcv36du)

**Reproduction Line:**
```
./gradlew ":libs:simdvec:test" --tests "org.elasticsearch.simdvec.ESVectorUtilTests.testSoarDistance" -Dtests.seed=CD8BF84AFCAAF96A -Dtests.locale=ebu-Latn-KE -Dtests.timezone=Asia/Aqtobe -Druntime.java=24
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.simdvec.ESVectorUtilTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSoarDistance'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<7750.9575> but was:<7750.9497>
```

**Issue Reasons:**
- [9.1] 2 failures in test testSoarDistance (1.7% fail rate in 119 executions)
- [9.1] 2 failures in pipeline elasticsearch-periodic-platform-support (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/libs/simdvec/src/test/java/org/elasticsearch/simdvec/ESVectorUtilTests.java
+++ b/libs/simdvec/src/test/java/org/elasticsearch/simdvec/ESVectorUtilTests.java
@@ -284,7 +284,7 @@ public class ESVectorUtilTests extends BaseVectorizationTests {
 
     public void testSoarDistance() {
         int size = random().nextInt(128, 512);
-        float deltaEps = 1e-5f * size;
+        float deltaEps = 1e-3f * size;
         var vector = new float[size];
         var centroid = new float[size];
         var preResidual = new float[size];

